### PR TITLE
feat(launch): lock_entrypoint field to launch agent config

### PR DESCRIPTION
**Pull request by bcsherma**
When the agent attempts to run a job
if the lock_entrypoint flag is set in the config
the agent will raise an error if any job it recieves attempts to override the entrypoint.

Description
-----------

Adds a field in the agent config to prevent the agent from running any launch spec with entrypoint overrides specified.

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at 00a38a9</samp>

> _Sing, O Muse, of the launch agent's mighty deed,_
> _How it locked the entrypoint with cunning skill,_
> _And checked the job specs with unerring heed,_
> _To guard the projects from all harm and ill._

### CODE REVIEW

File: wandb/sdk/launch/agent/agent.py [hunk 1]
```diff
@@ -301,6 +302,16 @@ def run_job(self, job: Dict[str, Any]) -> None:
         _logger.info("Parsing launch spec")
         launch_spec = job["runSpec"]
 
+        # Abort if this job attempts to override the entrypoint and we're not
+        # configured to allow that.
+        if self._lock_entrypoint:
+            overrides = launch_spec.get("overrides", {})
+            if overrides.get("entry_point"):
+                raise ValueError(
```
Comment (KyleGoyette) [inline wandb/sdk/launch/agent/agent.py:310]: Won't this fully crash the agent?

Comment (bcsherma) [reply wandb/sdk/launch/agent/agent.py:310]: No because the agent calls this function in a blanket try/except

Comment (bcsherma) [reply wandb/sdk/launch/agent/agent.py:310]: https://github.com/wandb/wandb/blob/00a38a95f1f824679810f7e05e6ccd1e70d0d9a0/wandb/sdk/launch/agent/agent.py#L359

**Comment by TimH98** [comment]
This can probably be closed/deleted since [secure mode](https://github.com/wandb/wandb/pull/5363) implements this

**Pull request closed without merging**